homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -83,6 +76,13 @@ async def _async_update_data(self) -> list[DataConnection]:
             )
             raise UpdateFailed from e
 
+        def calculate_duration_in_seconds(duration_text: str) -> int | None:
+            """Transform and calculate the duration into seconds."""
+            # Transform 01d03:21:23 into 01 days 03:21:23
+            duration_text_pg_format = duration_text.replace("d", " days ")
+            duration = dt_util.parse_duration(duration_text_pg_format)
+            return duration.seconds if duration else None
+
```

Comment:
Oh I meant, you can take it outside of the class altogether

Reply:
ohh, ok